pkg/controllers/networkpolicy/controller_test.go

```diff
@@ -433,36 +432,60 @@ var _ = Describe("NetworkPolicy", func() {
 		}
 	}
 
-	// The sequence of IPs in IngressRule/EgressRule might change during rule deduplication.
-	// Sort it for equality check.
-	sortRuleIPs := func(inputIP []*net.IPNet) []*net.IPNet {
-		if inputIP == nil {
-			return nil
+	checkRules := func(grpID *securitygroup.CloudResource, add, rm []*securitygroup.CloudRule) {
+		addIn := make([]securitygroup.IngressRule, 0)
+		addEg := make([]securitygroup.EgressRule, 0)
+		for _, rule := range add {
+			switch rule.Rule.(type) {
+			case *securitygroup.IngressRule:
+				addIn = append(addIn, *rule.Rule.(*securitygroup.IngressRule))
+			case *securitygroup.EgressRule:
+				addEg = append(addEg, *rule.Rule.(*securitygroup.EgressRule))
+			}
 		}
-		ipMap := make(map[string]*net.IPNet)
-		ipStrList := make([]string, 0)
-		for _, ip := range inputIP {
-			ipStrList = append(ipStrList, ip.String())
-			ipMap[ip.String()] = ip
+		rmIn := make([]securitygroup.IngressRule, 0)
+		rmEg := make([]securitygroup.EgressRule, 0)
+		for _, rule := range rm {
+			switch rule.Rule.(type) {
+			case *securitygroup.IngressRule:
+				rmIn = append(rmIn, *rule.Rule.(*securitygroup.IngressRule))
+			case *securitygroup.EgressRule:
+				rmEg = append(rmEg, *rule.Rule.(*securitygroup.EgressRule))
+			}
 		}
-		sort.Strings(ipStrList)
-		outputIP := make([]*net.IPNet, 0)
-		for _, ipStr := range ipStrList {
-			outputIP = append(outputIP, ipMap[ipStr])
+		list, _ := reconciler.cloudRuleIndexer.ByIndex(cloudRuleIndexerByAppliedToGrp, grpID.CloudResourceID.String())
+		currentIn := make([]securitygroup.IngressRule, 0)
+		currentEg := make([]securitygroup.EgressRule, 0)
+		for _, obj := range list {
+			rule := obj.(*securitygroup.CloudRule)
+			switch rule.Rule.(type) {
+			case *securitygroup.IngressRule:
+				currentIn = append(currentIn, *rule.Rule.(*securitygroup.IngressRule))
+			case *securitygroup.EgressRule:
+				currentEg = append(currentEg, *rule.Rule.(*securitygroup.EgressRule))
+			}
+		}
+		// number of current rules + number of adding rules - number of removing rules should equal to number of total target rules.
+		Expect(len(currentIn) + len(addIn) - len(rmIn)).To(Equal(len(ingressRule)))
+		Expect(len(currentEg) + len(addEg) - len(rmEg)).To(Equal(len(egressRule)))
+		// adding rules should not present in current rules and present it target rules.
+		for _, rule := range addIn {
```

Comment:
dont use short name.. addIngress and currentIngress

Reply:
Done